File: medplum-typescript/packages/sdk/src/model/fhir/CatalogEntry.ts
import { CatalogentryStatus } from "./CatalogentryStatus";
import { CatalogEntry_RelatedEntry } from "./CatalogEntry_RelatedEntry";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface CatalogEntry {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    type: CodeableConcept | null | undefined;
    orderable: boolean | null | undefined;
    referencedItem: Reference;
    additionalIdentifier: Identifier[] | null | undefined;
    classification: CodeableConcept[] | null | undefined;
    status: CatalogentryStatus | null | undefined;
    validityPeriod: Period | null | undefined;
    validTo: dateTime | null | undefined;
    lastUpdated: dateTime | null | undefined;
    additionalCharacteristic: CodeableConcept[] | null | undefined;
    additionalClassification: CodeableConcept[] | null | undefined;
    relatedEntry: CatalogEntry_RelatedEntry[] | null | undefined;
}

File: medplum-java/medplum-model/src/main/java/com/fhir/CatalogEntry.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCatalogEntry.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface CatalogEntry {
  Reference referencedItem();

  Optional<List<CatalogEntry_RelatedEntry>> relatedEntry();

  Optional<code> language();

  Optional<List<Extension>> modifierExtension();

  Optional<CodeableConcept> type();

  Optional<List<Identifier>> additionalIdentifier();

  Optional<Boolean> orderable();

  Optional<List<CodeableConcept>> additionalClassification();

  Optional<List<ResourceList>> contained();

  Optional<dateTime> validTo();

  String resourceType();

  Optional<List<Extension>> extension();

  Optional<CatalogentryStatus> status();

  Optional<id> id();

  Optional<uri> implicitRules();

  Optional<List<CodeableConcept>> classification();

  Optional<Narrative> text();

  Optional<Meta> meta();

  Optional<List<Identifier>> identifier();

  Optional<Period> validityPeriod();

  Optional<dateTime> lastUpdated();

  Optional<List<CodeableConcept>> additionalCharacteristic();

  static ImmutableCatalogEntry.ReferencedItemBuildStage builder() {
    return ImmutableCatalogEntry.builder();
  }
}
